Добавил запрос поэмоциям

Vitaly-Pon opened the pull request:
Добавил тест на сервис
Добавил обработку всех ошибок

Vitaly-Pon committed:
Добавил запрос поэмоциям
Добавил тест на сервис
Добавил обработку всех ошибок

AlexanderNest commented on core/src/main/java/com/example/demo/service/MovieRecommendationService.java:
```diff
@@ -24,7 +24,24 @@ public MovieRecommendationsResponse getMovieRecommendations(String genre, int co
-        var apiResponse = apiClient.getRecommendations(request);
+        DeepSeekChatResponse apiResponse = apiClient.getRecommendations(request);
+        String content = apiResponse.getChoices().get(0).getMessage().getContent();
+
```
дублирование кода, убрать

AlexanderNest commented on core/src/main/java/com/example/demo/util/PromptTemplates.java:
```diff
@@ -7,4 +7,11 @@ public static String movieRecommendationPrompt(String genre, int count){
+
+    public static String movieRecommendationOnEmotionPrompt(String emotion, int count){
+        return String.format(
+                "Порекомендуй ровно %d лучших фильмов с настроением '%s'. " +
```
плохая работа со строками, изменить

AlexanderNest commented on core/src/main/java/com/example/demo/util/PromptTemplates.java:
```diff
@@ -7,4 +7,11 @@ public static String movieRecommendationPrompt(String genre, int count){
                 "Без вводных слов.", count, genre);
     }
+
+    public static String movieRecommendationOnEmotionPrompt(String emotion, int count){
```
плохое название метода

AlexanderNest commented on core/src/test/com/example/demo/MovieRecommendationServiceTest.java:
```diff
@@ -0,0 +1,52 @@
+import org.mockito.Mock;
+
+@ExtendWith(MockitoExtension.class)
+public class MovieRecommendationServiceTest {
```
тест не запускается. рекомендую заново создать проект с зависимостями как в основном проекте

AlexanderNest commented on core/src/main/java/com/example/demo/controller/MovieRecommendationController.java:
```diff
@@ -21,8 +21,13 @@ public MovieRecommendationController(MovieRecommendationService movieRecommendat
 
-    @PostMapping("/recommend")
-    public MovieRecommendationsResponse getRecommendMovies(@Valid @ModelAttribute MovieRecommendationRequest request) {
+    @PostMapping("/recommendInGenre")
```
это вынести в интерфейс

AlexanderNest commented on core/src/main/java/com/example/demo/controller/MovieRecommendationController.java:
```diff
@@ -21,8 +21,13 @@ public MovieRecommendationController(MovieRecommendationService movieRecommendat
         this.movieRecommendationService = movieRecommendationService;
```
используем lombok

Vitaly-Pon committed:
Добавил запрос поэмоциям
Добавил тест на сервис
Добавил обработку всех ошибок
fix

Vitaly-Pon committed:
Добавил тест на сервис

Vitaly-Pon committed:
Убрал дублирование кода из сервиса
Создал интерфейс сервиса
@AllArgsConstructor в сервис и контроллер
В PromptTemplates добавил константы шаблоны

Vitaly-Pon committed:
Убрал дублирование кода из сервиса
Создал интерфейс сервиса
@AllArgsConstructor в сервис и контроллер
В PromptTemplates добавил константы шаблоны
Лишние импорты убраны

Vitaly-Pon committed:
Убрал дублирование кода из сервиса
Создал интерфейс сервиса
@AllArgsConstructor в сервис и контроллер
В PromptTemplates добавил константы шаблоны
Убрал лишнее
